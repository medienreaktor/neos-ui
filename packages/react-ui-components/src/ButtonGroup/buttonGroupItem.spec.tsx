import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';

import ButtonGroupItem, {ButtonGroupItemProps} from './buttonGroupItem';

describe('<ButtonGroupItem/>', () => {
    const props: ButtonGroupItemProps = {
        element: <div id="foo">Foo button</div>,
        id: 'foo',
        onClick: jest.fn(),
    };

    it('should render correctly.', () => {
        const wrapper = shallow(<ButtonGroupItem {...props}/>);

        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should allow the propagation of "className" with the "className" prop.', () => {
        const wrapper = shallow(<ButtonGroupItem {...props} className="fooClassName"/>);

        expect(wrapper.prop('className')).toContain('fooClassName');
    });
});
