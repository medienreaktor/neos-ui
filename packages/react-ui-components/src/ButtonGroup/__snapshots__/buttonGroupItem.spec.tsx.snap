// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonGroupItem/> should render correctly. 1`] = `
<div
  id="foo"
  onClick={[Function]}
>
  Foo button
</div>
`;
